govcd/ejecttask.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+type EjectTask struct {
+	*Task
+	vm *VM
+}
+
+// Creates wrapped Task which is dedicated for eject media functionality and
+// provides additional functionality to answer VM questions
+func NewEjectTask(task *Task, vm *VM) *EjectTask {
+	return &EjectTask{
+		task,
+		vm,
+	}
+}
+
+// Checks the status of the task every 3 seconds and returns when the
+// eject task is either completed or failed
+func (ejectTask *EjectTask) WaitTaskCompletion(isAnswerYes bool) error {
+	return ejectTask.WaitInspectTaskCompletion(isAnswerYes, 3*time.Second)
```

Comment:
Please expose this retry interval variable at the top of the file, so everyone notices that there's some waiting game involved.
Also, if the call is not too expensive let's make it check every single second as opposed to three.

Reply:
Changed